src/python/python/xml.py

```diff
@@ -0,0 +1,617 @@
+        name = self.file_stack[self.current_file].pop()
+
+        self.wf(self.current_file, '</%s>\n' % name, self.file_tabs[self.current_file])
+
+    def element(self, name, attributes={}, file=None):
+        if file is not None:
+            self.set_output_file(file)
+
+        self.wf(self.current_file, '<%s' % name, self.file_tabs[self.current_file])
+
+        for (k, v) in attributes.items():
+            self.wf(self.current_file, ' %s=\"%s\"' % (k, v))
+
+        self.wf(self.current_file, '/>\n')
+
+    def parameter(self, param_type, param_name, attributes={}, file=None):
+        if file is not None:
+            self.set_output_file(file)
+
+        self.wf(self.current_file, '<%s name="%s"' % (param_type, param_name), self.file_tabs[self.current_file])
+
+        for (k, v) in attributes.items():
+            self.wf(self.current_file, ' %s=\"%s\"' % (k, v))
+
+        self.wf(self.current_file, '/>\n')
+
+    def get_plugin_tag(self, plugin_type):
+        from mitsuba import variant
+        class_ =  self.pmgr.get_plugin_class(plugin_type, variant())
+        if not class_: # if get_plugin_class returns None, there is not corresponding plugin
+            return None
+        class_ = class_.parent()
+        while class_.alias() == class_.name():
+            class_ = class_.parent()
+        return class_.alias()
+
+    def current_tag(self):
+        return self.file_stack[self.current_file][-1]
+
+    def configure_defaults(self, scene_dict):
+        '''
+        Traverse the scene graph and look for properties in the defaults dict.
+        For such properties, store their value in a default tag and replace the value by $name in the prop.
+        '''
+        for key, value in scene_dict.items():
+            if isinstance(value, dict):
+                self.configure_defaults(value)
+            elif key in self.defaults:
+                if '$%s'%self.defaults[key] in self.scene_data[Files.MAIN]:
+                    print("****** Already exported default for : %s ******" % key)
+                    continue
+                params = {
+                    'type': 'default',
+                    'name': self.defaults[key],
+                    'value': value
+                }
+                self.data_add('$%s'%self.defaults[key], params)
+                if isinstance(value, int):
+                    scene_dict[key] = {'type': 'integer'}
+                elif isinstance(value, float):
+                    scene_dict[key] = {'type': 'float'}
+                elif isinstance(value, str):
+                    scene_dict[key] = {'type': 'string'}
+                elif isinstance(value, bool):
+                    scene_dict[key] = {'type': 'boolean'}
+                else:
+                    raise ValueError("Unsupported default type: %s" % value)
+                #TODO: for now, the only supported defaults are ints, so that works. This may not always be the case though
```

Comment:
Would be great to support other types as well!

Reply:
Well the question is more: which defaults do we automatically add? For now, the defaults added are all ints